File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        stack = []

        for ch in s:
            if(not stack):
                stack.append(ch)
                continue
            if( (stack[-1] == '(' and ch == ')') or (stack[-1] == '{' and ch == '}') or(stack[-1] == '[' and ch == ']') ):
                stack.pop()
            else:
                stack.append(ch)
        
        return len(stack) == 0            

File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        
        stack<char> stack;

        for(char ch : s){

            if(stack.empty()){
                stack.push(ch);
                continue;
            }

            if(
                (stack.top() == '(' && ch == ')') ||
                (stack.top() == '{' && ch == '}') ||
                (stack.top() == '[' && ch == ']')
            ){
                stack.pop();
            }
            else{
                stack.push(ch);
            }

        }

        return stack.empty();

    }
};